src/java.base/share/classes/jdk/internal/util/PlatformProps.java.template

```diff
@@ -0,0 +1,93 @@
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.internal.util;
+
+import java.util.Locale;
+
+/**
+ * The corresponding source file is generated by GensrcMisc.gmk for java.base.
+ * @see OperatingSystem
+ * @see Architecture
+ */
+class PlatformProps {
+
+    // Index/ordinal of the current OperatingSystem enum as substituted by the build
+    static final OperatingSystem CURRENT_OS = initOS("@@OPENJDK_TARGET_OS@@");
+
+    // Precomputed booleans for each Operating System
+    static final boolean TARGET_OS_IS_LINUX   = "@@OPENJDK_TARGET_OS@@" == "linux";
+    static final boolean TARGET_OS_IS_MACOS   = "@@OPENJDK_TARGET_OS@@" == "macosx";
+    static final boolean TARGET_OS_IS_WINDOWS = "@@OPENJDK_TARGET_OS@@" == "windows";
+    static final boolean TARGET_OS_IS_AIX     = "@@OPENJDK_TARGET_OS@@" == "aix";
+
+    /**
+     * Returns the OperatingSystem of the build.
+     * Build time names are mapped to respective uppercase enum values.
+     * Names not recognized are throw an ExceptionInInitializerError with IllegalArgumentException.
+     */
+    private static OperatingSystem initOS(String osName) {
+        String mapped = switch (osName) {
+            case "macosx" -> "macos";
+            default -> osName;
+        };
+        return OperatingSystem.valueOf(mapped.toUpperCase(Locale.ROOT));
+    }
+
+    // The Architecture value for the current architecture
+    static final Architecture CURRENT_ARCH = initArch("@@OPENJDK_TARGET_CPU@@");
+
+    // Architecture.is64Bit() uses this value
+    static final int TARGET_ARCH_BITS = @@OPENJDK_TARGET_CPU_BITS@@;
+
+    // Architecture.isLittleEndian value from the build
+    static final boolean TARGET_ARCH_LITTLE_ENDIAN = "@@OPENJDK_TARGET_CPU_ENDIAN@@" == "little";
+
+    // Precomputed booleans for each Architecture, shared with jdk.internal.util.Architecture
+    // The variables are named to match the Architecture value names, and
+    // the values are named to match the build variables.
+    static final boolean TARGET_ARCH_IS_X64         = "@@OPENJDK_TARGET_CPU@@" == "x86_64";
+    static final boolean TARGET_ARCH_IS_X86         = "@@OPENJDK_TARGET_CPU@@" == "x86";
+    static final boolean TARGET_ARCH_IS_AARCH64     = "@@OPENJDK_TARGET_CPU@@" == "aarch64";
+    static final boolean TARGET_ARCH_IS_RISCV64     = "@@OPENJDK_TARGET_CPU@@" == "riscv64";
+    static final boolean TARGET_ARCH_IS_S390        = "@@OPENJDK_TARGET_CPU@@" == "s390x";
+    static final boolean TARGET_ARCH_IS_PPC64       = "@@OPENJDK_TARGET_CPU@@" == "ppc64";
+
+    /**
+     * Returns the Architecture of the built architecture.
+     * Build time names are mapped to respective uppercase enum values.
+     * Names not recognized are mapped to Architecture.OTHER.
+     */
+    private static Architecture initArch(String archName) {
+        try {
+            String mapped = switch (archName) {
```

Comment:
How about mapping names in the build script instead of at runtime?

I suggest placing only the mapped arch name in `PlatformProps` and moving the `CURRENT_ARCH` to `Architecture`.